src/compute/src/render/flat_map.rs

```diff
@@ -33,11 +33,19 @@ where
         let (ok_collection, err_collection) = input.as_specific_collection(input_key.as_deref());
         let (oks, errs) = ok_collection.inner.flat_map_fallible("FlatMapStage", {
             let mut datums = DatumVec::new();
+            let mut datums_mfp = DatumVec::new();
+
             move |(input_row, mut time, diff)| {
+                let mut table_func_output = Vec::new();
+
+                // Into which we accumulate result update triples.
+                let mut oks_output = Vec::new();
+                let mut err_output = Vec::new();
```

Comment:
Follow-up: Move the allocations to once per invocation instead of once per input row.